contracts/contracts/strategies/sonic/SonicValidatorDelegator.sol

```diff
@@ -0,0 +1,363 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { IERC20, InitializableAbstractStrategy } from "../../utils/InitializableAbstractStrategy.sol";
+import { IVault } from "../../interfaces/IVault.sol";
+import { ISFC } from "../../interfaces/sonic/ISFC.sol";
+import { IWrappedSonic } from "../../interfaces/sonic/IWrappedSonic.sol";
+
+/**
+ * @title Manages delegation to Sonic validators
+ * @notice This contract implements all the required functionality to delegate to,
+   undelegate from and withdraw from validators.
+ * @author Origin Protocol Inc
+ */
+abstract contract SonicValidatorDelegator is InitializableAbstractStrategy {
+    /// @notice Address of Sonic's wrapped S token
+    address public immutable wrappedSonic;
+    /// @notice Address of Sonic's Special Fee Contract (SFC)
+    address public immutable sfc;
```

Comment:
⚪️ If we set the variable type here to `ISFC`, it should remove almost ten casts from the code base, at the cost of only one address() cast the other way. Would make the code read cleaner.

Reply:
good idea 👍